src/dump.c

```diff
@@ -652,28 +714,51 @@ static void jl_serialize_value_(jl_serializer_state *s, jl_value_t *v, int as_li
         jl_array_t *values = (jl_array_t*)jl_fieldref_noalloc(v, 0);
         size_t l = jl_array_len(values);
         if (l <= 255) {
-            writetag(s->s, jl_phicnode_type);
+            write_uint8(s->s, TAG_PHICNODE);
             write_uint8(s->s, (uint8_t)l);
         }
         else {
-            writetag(s->s, (jl_value_t*)LongPhic_tag);
+            write_uint8(s->s, TAG_LONG_PHICNODE);
             write_int32(s->s, l);
         }
         for (i = 0; i < l; i++) {
             jl_serialize_value(s, jl_array_ptr_ref(values, i));
         }
     }
+    else if (jl_is_gotonode(v)) {
+        write_uint8(s->s, TAG_GOTONODE);
+        jl_serialize_value(s, jl_get_nth_field(v, 0));
+    }
+    else if (jl_is_quotenode(v)) {
```

Comment:
Perhaps should have `&& (jl_isbits(v.value) || !is_ast_node(v.value))` here?

Reply:
I don't think that's necessary, since all this really does is shorten the object header for a QuoteNode.

Reply:
It makes sure that the field doesn't have a special representation (which could end up creating (non-egal) copies of the supposed-to-be-constant object in the AST)

Reply:
Ah --- that check would have to be added in `is_ast_node`. At this point we're definitely serializing it either way.

Reply:
Wait, that still doesn't matter since the value will be recursively checked for `is_ast_node` anyway.